### PR TITLE
Write separate completion flags for each job

### PR DESCRIPTION
**Pull request by pauldwolfe**

### CODE REVIEW

File: bam/src/main/java/com/hartwig/bam/after/BamIndexPipeline.java [hunk 1]
```diff
@@ -31,24 +33,31 @@ private BamIndexPipeline(final FileSystem fileSystem, final String bamFolder, fi
         this.sortAndIndexer = sortAndIndexer;
     }
 
-    public void execute(Sample sample) throws IOException, InterruptedException {
-        String bamFileLocation = String.format("/%s/%s.bam", sourceBamDirectory, sample.name()).substring(1);
-        String unsortedBam = Bams.name(sample, localBamDirectory, "unsorted");
+    public void execute(Sample sample, StatusReporter statusReporter) {
```
Comment (pauldwolfe) [inline bam/src/main/java/com/hartwig/bam/after/BamIndexPipeline.java]: Why pass the reporter as an argument here when we use it as a field in the other two?

File: cluster/src/main/java/com/hartwig/pipeline/execution/dataproc/GoogleDataproc.java [hunk 2]
```diff
@@ -109,9 +110,15 @@ private String jobIdAndClusterName(final RuntimeBucket runtimeBucket, final Spar
         return untrimmed.substring(0, Math.min(untrimmed.length(), 50));
     }
 
-    private boolean isDone(final String jobName, final RuntimeBucket runtimeBucket) {
-        List<StatusCheck.Status> finalStatuses = asList(StatusCheck.Status.SUCCESS, StatusCheck.Status.FAILED);
-        return finalStatuses.contains(getStatus(jobName, runtimeBucket));
+    private boolean isDone(final Job job, final String jobName, final RuntimeBucket runtimeBucket) {
+        LOGGER.info("Checking if job is done");
+        String state = job.getStatus().getState();
+        if (state.equals("DONE")) {
```
Comment (pauldwolfe) [inline cluster/src/main/java/com/hartwig/pipeline/execution/dataproc/GoogleDataproc.java]: Not sure this naming really correct. Isn't a"CANCELLED" or "ERROR" job "done"? Maybe isComplete is a better name?

File: cluster/src/main/java/com/hartwig/pipeline/execution/dataproc/GoogleDataproc.java [hunk 3]
```diff
@@ -221,7 +228,7 @@ private void waitForOperationComplete(Operation operation) throws IOException {
         boolean operationComplete = isDone.test(operation);
         while (!operationComplete) {
             sleep();
-            LOGGER.debug("Operation [{}] not complete, waiting...", description.apply(operation));
+            LOGGER.info("Operation [{}] not complete, waiting...", description.apply(operation));
```
Comment (pauldwolfe) [inline cluster/src/main/java/com/hartwig/pipeline/execution/dataproc/GoogleDataproc.java]: did you mean to commit this?

Comment (nedleitch) [reply cluster/src/main/java/com/hartwig/pipeline/execution/dataproc/GoogleDataproc.java]: I did not. Next commit will revert it.

**Pull request merged**